pandas/_libs/tslib.pyx

```diff
@@ -345,6 +349,18 @@ def array_with_unit_to_datetime(
                 if val != val or val == NPY_NAT:
                     iresult[i] = NPY_NAT
                 else:
+                    if is_ym and is_float_object(val) and val != int(val):
```

Comment:
nothing major, but I think `val % 1` would be more performant (and IMO clearer)
```python
In [9]: %%timeit
   ...: if val != int(val): pass
   ...: 
   ...: 
67.4 ns ± 4.53 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [10]: %%timeit
    ...: if val % 1: pass
    ...: 
    ...: 
34.8 ns ± 1.49 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Reply:
updated to use `val.is_integer()`, maximally clear, _almost_ as performant as mod.

Reply:
nice!